lib/balance-tx/lib/internal/Internal/Cardano/Write/Tx.hs

```diff
@@ -337,6 +337,7 @@ type RecentEraLedgerConstraints era =
     , Alonzo.AlonzoEraTxBody era
     , Babbage.ShelleyEraTxBody era
     , ScriptsNeeded era ~ AlonzoScriptsNeeded era
+    , Babbage.BabbageEraTxBody era
```

Comment:
I've made a commit that simplifies these constraints here: c06fb3fe7fd8b11dc16fb0c79767ad7f8d9f37c1

(`{Shelley,Alonzo}EraTxBody` are both implied by `BabbageEraTxBoday`, so we ought to be able to remove them.)